var/spack/repos/builtin/packages/microsocks/package.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Microsocks(MakefilePackage):
+    """Microsocks is a multithreaded, small, efficient SOCKS5 server.
+    It is a SOCKS5 service that you can run on your remote boxes to
+    tunnel connections through them, if for some reason SSH doesn't
+    cut it for you."""
+
+    homepage = "https://github.com/rofl0r/microsocks"
+    url      = "https://github.com/rofl0r/microsocks/archive/refs/tags/v1.0.2.tar.gz"
+    git      = "https://github.com/rofl0r/microsocks.git"
+
+    maintainers = ["jcpunk"]
+
+    version('develop', branch='master')
+    version('1.0.2', sha256='5ece77c283e71f73b9530da46302fdb4f72a0ae139aa734c07fe532407a6211a')
+
+    depends_on('gcc languages=c', type="build")
```

Comment:
If this is intended to control the compiler used to build `microsocks`, then FYI this his won't work as intended: Spack treats compilers differently than other dependencies.

If `gcc` is the only supported compiler for `microsocks`, then currently the only way to enforce that is to add something like:

```
conflicts('%intel')
conflicts('%clang')
...
# one line for each compiler, you may be able to generate this by extracting names in lib/spack/spack/compilers/
```

Or (since this is onerous) you can also add a check at the beginning of `build()` which raises an exception if `not spec.satisfies('%gcc')` (this doesn't affect the concretization though).

We have plans to support using `depends_on(compiler)` in that manner in the future.

Reply:
I'll investigate if the other C compilers work with this code base.

Is there a way I can ensure access to a C compiler or is that implicit because it is a source build?

Reply:
It's implied, so the way you have it now is good.